service/history/replication/task_processor.go

```diff
@@ -540,6 +551,61 @@ func (p *taskProcessorImpl) generateDLQRequest(
 	}
 }
 
+func (p *taskProcessorImpl) triggerDataInconsistencyScan(replicationTask *types.ReplicationTask) error {
+
+	var failoverVersion int64
+	var domainID string
+	var workflowID string
+	var runID string
+	switch {
+	case replicationTask.GetHistoryTaskV2Attributes() != nil:
+		attr := replicationTask.GetHistoryTaskV2Attributes()
+		versionHistoryItems := attr.GetVersionHistoryItems()
+		if versionHistoryItems == nil || len(versionHistoryItems) == 0 {
+			return errors.New("failed to trigger data scan due to invalid version history")
+		}
+		// version history items in same batch should be the same
+		failoverVersion = versionHistoryItems[0].GetVersion()
+		domainID = attr.GetDomainID()
+		workflowID = attr.GetWorkflowID()
+		runID = attr.GetRunID()
+	case replicationTask.GetSyncActivityTaskAttributes() != nil:
+		attr := replicationTask.GetSyncActivityTaskAttributes()
+		failoverVersion = replicationTask.GetSyncActivityTaskAttributes().Version
+		domainID = attr.GetDomainID()
+		workflowID = attr.GetWorkflowID()
+		runID = attr.GetRunID()
+	default:
+		return nil
+	}
+	clusterName := p.shard.GetClusterMetadata().ClusterNameForFailoverVersion(failoverVersion)
+	client := p.shard.GetService().GetClientBean().GetRemoteFrontendClient(clusterName)
+	fixExecution := entity.Execution{
+		DomainID:   domainID,
+		WorkflowID: workflowID,
+		RunID:      runID,
+		State:      persistence.WorkflowStateCorrupted,
```

Comment:
nit: mind add a comment here that says we assume it's corrupted here and rely on the check before fix logic in the fixer workflow to decide if it's actually a corruption?